utils/libtrixi-init-julia

```diff
@@ -335,14 +335,6 @@ fi
 [ $? -eq 0 ] || die "could not configure system HDF5 library for Julia"
 echo
 
-# Develop LibTrixi.jl
-echo "Install LibTrixi.jl..."
-JULIA_DEPOT_PATH="$julia_depot" JULIA_PKG_PRECOMPILE_AUTO=0 \
-    $julia_exec --project=. \
-    -e "using Pkg; Pkg.develop(path=\"$libtrixi_jl_path\")"
-[ $? -eq 0 ] || die "could not install LibTrixi.jl"
-echo
```

Comment:
Is this change even necessary now that you use the newer t8code version?

If yes, I recommend to expand the comment in the line below.

Reply:
:thinking: 

I would argue that at his point during setup the _external_ t8code library does not play a role, yet.

But apart from that: This segfault made me try a different t8code version
https://github.com/trixi-framework/libtrixi/actions/runs/9503077644/job/26192654449#step:22:903

Now I am thinking about compatibility of T8code.jl and the external t8code library: Trixi.jl specifies compat bounds for T8code.jl, but how about T8code.jl and t8code when not using jlls?

Reply:
When using external libraries, there's just no way to specify compat bounds. We can - and do - require compat bounds on the JLL, see
https://github.com/DLR-AMR/T8code.jl/blob/5c9b43f95c703df8bb06f90c683717c64b42b985/Project.toml#L25
Other than that, I think it's up to the user to do the right thing...

Reply:
Good point! This gives an idea on which version to use.

So, what's missing here?

Reply:
> So, what's missing here?

What do you mean by that?

Reply:
Sorry!

> Is this change even necessary now that you use the newer t8code version?
Yes.

> If yes, I recommend to expand the comment in the line below.
Do you think we should add anything else in this PR (besides your explaining comment)?

Reply:
Nope, all good then, I guess 😊